### PR TITLE
Avoid a bunch of warnings in release mode.

### LINKED ISSUE #9609: face-based loop warnings
I am getting a bunch of warnings as a result of #9603 (?):
```

In file included from /home/heister/deal-git/source/fe/fe_tools_interpolate.cc:17:
/home/heister/deal-git/include/deal.II/fe/fe_tools_interpolate.templates.h:179:37: warning: unused variable 'face' [-Wunused-variable]
            for (const unsigned int face : GeometryInfo<dim>::face_indices())
                                    ^
/home/heister/deal-git/include/deal.II/fe/fe_tools_interpolate.templates.h:335:37: warning: unused variable 'face' [-Wunused-variable]
            for (const unsigned int face : GeometryInfo<dim>::face_indices())
                                    ^
/home/heister/deal-git/include/deal.II/fe/fe_tools_interpolate.templates.h:670:37: warning: unused variable 'face' [-Wunused-variable]
            for (const unsigned int face : GeometryInfo<dim>::face_indices())
                                    ^
3 warnings generated.
```

@bangerth can you take a look?

This is with clang 6.0

### PR DESCRIPTION
**Pull request by bangerth**
Fixes #9609, I believe. At least I don't see any more warnings after this patch when compiling the release library.

/rebuild

### CODE REVIEW

File: include/deal.II/fe/fe_tools_interpolate.templates.h [hunk 1]
```diff
@@ -175,12 +175,13 @@ namespace FETools
             ((cell2->get_fe().dofs_per_vertex != 0) &&
              (constraints.n_constraints() == 0));
 
+#if DEBUG
```
Comment (kronbichler) [inline include/deal.II/fe/fe_tools_interpolate.templates.h]: While not wrong, I believe we most often write `#ifdef DEBUG`.

**Comment by bangerth** [comment]
Fixed.

File: include/deal.II/fe/fe_tools_interpolate.templates.h [hunk 2]
```diff
@@ -666,11 +655,19 @@ namespace FETools
       if ((cell->subdomain_id() == subdomain_id) ||
           (subdomain_id == numbers::invalid_subdomain_id))
         {
+#ifdef DEBUG
+          // For continuous elements on grids with hanging nodes we need
+          // hanging node constraints. Consequently, when the elements are
+          // continuous no hanging node constraints are allowed.
+          const bool hanging_nodes_not_allowed =
+            (dof1.get_fe().dofs_per_vertex != 0) || (fe2.dofs_per_vertex != 0);
```
Comment (masterleinad) [inline include/deal.II/fe/fe_tools_interpolate.templates.h]: I guess it's fine setting this flag for every iteration anew. First, it's only in `Debug` mode and second, it is so much cheaper than everything else we are doing here.

Comment (bangerth) [reply include/deal.II/fe/fe_tools_interpolate.templates.h]: That was my thinking. The variable is unused outside the assertion, so I figured that it's worth moving it under the `#ifdef`.

**Pull request merged**